packages/edit-site/src/store/actions.js

```diff
@@ -236,3 +242,68 @@ export function updateSettings( settings ) {
 		settings,
 	};
 }
+
+/**
+ * Reverts a template to its original theme-provided file.
+ *
+ * @param {Object} template The template to revert.
+ */
+export function* revertTemplate( template ) {
+	if ( ! isTemplateRevertable( template ) ) {
+		yield controls.dispatch(
+			'core/notices',
+			'createErrorNotice',
+			__( 'This template is not revertable.' ),
+			{ type: 'snackbar' }
+		);
+		return;
+	}
+
+	try {
+		yield controls.dispatch(
+			'core',
+			'deleteEntityRecord',
+			'postType',
+			'wp_template',
+			template.id
+		);
+
+		const fileTemplate = yield controls.resolveSelect(
+			'core',
+			'getEntityRecord',
+			'postType',
+			'wp_template',
+			template.id
+		);
+
+		if ( ! fileTemplate ) {
+			yield controls.dispatch(
+				'core/notices',
+				'createErrorNotice',
+				__(
+					'The editor has encountered an unexpected error. Please reload.'
+				),
+				{ type: 'snackbar' }
+			);
+			return;
+		}
+
```

Comment:
Refreshing the `wp_template` records is apparently not enough, because the editor automatically loads the record's `edits`, rather than its content.

As far as I could understand, this is what happens:
- Open Site Editor at the `tt1-blocks//front-page` template.
- Once the entity is provided to the editor, a new `edits` record for `tt1-blocks//front-page` is created.
- Modify the template and save.
- The `edits` record for the template is updated with the latest change.
- Revert the template.
- The `tt1-blocks//front-page` entity record is deleted, and replaced by the original file version.
- The editor goes into "loading".
- Then it displays the `tt1-blocks//front-page` edits, which are still those belonging to the deleted entity record.

I couldn't find a way to wipe the `edits`, and I'm not even sure I _should_ wipe it, as I might mess with the undo stack.

I wonder what would be the most appropriate way to get the editor to ignore stale edits, if the entity record has been replaced by a newer/different version. 🤔 
Maybe I should manually `editEntityRecords` with the original file's content?

Reply:
The ID is the same. So in this case it makes sense. I think it would make sense to manually do it. We could even undo the revert in that case? I mean, the content change, not the actual revert. Which is a great thing. Maybe someone reverts the template then realizes it's a mistake.

Reply:
```js
import { parse } from '@wordpress/blocks';
........
yield controls.dispatch(
	'core',
	'editEntityRecord',
	'postType',
	'wp_template',
	fileTemplate.id,
	{
		blocks: parse( fileTemplate?.content?.raw ),
	}
);
```

Reply:
This works too but it has a lightly different meaning, that said, It's better than the hack to delete and fetch again.

Reply:
The problem with the last proposal here though is the the "template" stays revertable even if it was just reverted. Maybe it's fine though.